internal/admin/handle_authorized_apps.go

```diff
@@ -0,0 +1,215 @@
+			priorKey := form.PriorKey()
+
+			log.Printf("Deleting authorized app: %v", priorKey)
+			if err := aadb.DeleteAuthorizedApp(ctx, priorKey); err != nil {
+				ErrorPage(c, fmt.Sprintf("Error deleting authorized app: %v", err))
+				return
+			}
+
+			authorizedApp := model.NewAuthorizedApp()
+			if err := addHealthAuthorityInfo(ctx, verdb.New(s.env.Database()), authorizedApp, m); err != nil {
+				ErrorPage(c, err.Error())
+				return
+			}
+
+			m.AddSuccess(fmt.Sprintf("Successfully deleted app `%v`", priorKey))
+			m["app"] = authorizedApp
+			c.HTML(http.StatusOK, "authorizedapp", m)
+			return
+		}
+
+		ErrorPage(c, "Invalid request, app to edit not found.")
+	}
+}
+
+// HandleAuthorizedAppsShow handles the show page for authorized apps.
+func (s *Server) HandleAuthorizedAppsShow() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+		m := TemplateMap{}
+
+		appID, _ := c.GetQuery("apn")
+		authorizedApp := model.NewAuthorizedApp()
+
+		if appID == "" {
+			m["new"] = true
+		} else {
+			aadb := database.New(s.env.Database())
+			var err error
+			authorizedApp, err = aadb.GetAuthorizedApp(ctx, appID)
+			if err != nil {
+				ErrorPage(c, err.Error())
+				return
+			}
+		}
+
+		// Load the health authorities.
+		if err := addHealthAuthorityInfo(ctx, verdb.New(s.env.Database()), authorizedApp, m); err != nil {
+			ErrorPage(c, err.Error())
+			return
+		}
+
+		m["app"] = authorizedApp
+		m["previousKey"] = base64.StdEncoding.EncodeToString([]byte(authorizedApp.AppPackageName))
+		c.HTML(http.StatusOK, "authorizedapp", m)
+	}
+}
+
+// addHealthAuthorityInfo is a helper that adds HA info to the template map.
+func addHealthAuthorityInfo(ctx context.Context, haDB *verdb.HealthAuthorityDB, app *model.AuthorizedApp, m TemplateMap) error {
+	// Load the health authorities.
+	healthAuthorities, err := haDB.ListAllHealthAuthoritiesWithoutKeys(ctx)
+	if err != nil {
+		return fmt.Errorf("error loading health authorities: %w", err)
+	}
+
+	usedAuthorities := make(map[int64]bool)
+	for _, k := range app.AllAllowedHealthAuthorityIDs() {
+		usedAuthorities[k] = true
+	}
+
+	m["has"] = healthAuthorities
+	m["usedHealthAuthorities"] = usedAuthorities
+	return nil
+}
+
+type authorizedAppFormData struct {
+	// Top Level
+	FormKey string `form:"key"`
+	Action  string `form:"action"`
+
+	// Authorized App Data
+	AppPackageName                    string  `form:"app-package-name"`
+	AllowedRegions                    string  `form:"regions"`
+	BypassHealthAuthorityVerification bool    `form:"bypass-health-authority-verification"`
+	BypassRevisionToken               bool    `form:"bypass-revision-token"`
+	HealthAuthorityIDs                []int64 `form:"health-authorities"`
+}
+
+func (f *authorizedAppFormData) PriorKey() string {
+	if f.FormKey != "" {
+		bytes, err := base64.StdEncoding.DecodeString(f.FormKey)
+		if err != nil {
+			return ""
+		}
+		return string(bytes)
+	}
+	return ""
```

Comment:
Minor nit on inversion order to reduce nesting in the more critical case
```go
if f.FormKey == "" {
     return ""
}

bs, err := base64...
...
return string(bytes)
```

Reply:
This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.